src/types.ts

```diff
@@ -88,3 +86,18 @@ export enum VisualState {
   animatingOut = "animating-out",
   hidden = "hidden",
 }
+
+export interface IHistoryItem {
+  perform: () => any;
+  negate: () => any;
+}
+
+export interface IHistory {
+  undoStack: IHistoryItem[];
+  redoStack: IHistoryItem[];
+  add: (item: IHistoryItem) => IHistoryItem;
+  remove: (item: IHistoryItem) => void;
+  undo: (item?: IHistoryItem) => void;
+  redo: (item?: IHistoryItem) => void;
```

Comment:
Todo: `History` should probably only store and retrieve items; we should leave the `undo`/`redo` to the implementer, which in our case is the `Command` object.